qiskit/pulse/ir/pulse_ir.py

```diff
@@ -0,0 +1,400 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=cyclic-import
+
+"""
+=========
+Pulse IR
+=========
+
+"""
+
+from typing import Union, Optional, List, Set, Dict
+
+import numpy as np
+
+from qiskit.pulse.exceptions import PulseError
+
+from qiskit.pulse import (
+    LogicalElement,
+    Frame,
+    MixedFrame,
+    Qubit,
+)
+
+from qiskit.pulse.transforms import AlignmentKind, AlignSequential
+
+from .ir_instructions import (
+    BaseIRInstruction,
+    GenericInstruction,
+    AcquireInstruction,
+)
+
+
+class PulseIR:
+    """
+    ``PulseIR`` is the backbone of the  intermediate representation used in the Qiskit Pulse compiler.
+    A pulse program is represented as a single ``PulseIR`` object, with elements
+    which include IR instructions (base class :class:`~.BaseIRInstruction) and other
+    nested ``PulseIR`` objects. This structure mimics that of :class:`.qiskit.pulse.ScheduleBlock`
+    which is the main pulse program format used in Qiskit.
+
+    The IR is ment to support compilation needs, including partial instructions broadcasting,
+    scheduling, validation and more.
+    """
```

Comment:
Is this a module, a program, a sequence, a block? I believe the name `PulseIR` could be more descriptive.

Reply:
Thanks. I will change that.